src/com/skillsup/view/ProductMenu.java

```diff
@@ -31,30 +40,84 @@ public void show() {
+                    ifContinueMenu(scanner);
                     break;
-
                 case "2":
                     productDao
                             .getAllProductz()
                             .forEach(System.out::println);
+                    ifContinueMenu(scanner);
+                    break;
+                case "3":
+                    System.out.println("Input search category:");
+                    String category = scanner.nextLine();
+                    productDao
+                            .getByCategoryName(category)
+                            .forEach(System.out::println);
+                    ifContinueMenu(scanner);
                     break;
                 case "4":
-                    System.out.println("Input search string:");
+                    System.out.println("Input search product:");
                     String name = scanner.nextLine();
                     productDao
-                            .getByProductSubstring(name)
+                            .getByNameSubstring(name)
                             .forEach(System.out::println);
+                    ifContinueMenu(scanner);
                     break;
                 case "5":
                     BigDecimal min = BigDecimal.valueOf(getDoubleInput("input min price:", scanner)) ;
                     BigDecimal max = BigDecimal.valueOf(getDoubleInput("input max price:", scanner));
                     productDao
                             .filterProductsByPrice(min, max)
                             .forEach(System.out::println);
+                    ifContinueMenu(scanner);
+                    break;
+                case "6":
+//                    if (!currentUser.getUserRole().equals(UserRole.ADMIN)) {
+//                    System.out.println("Sorry, You do not have the authority for this operation! ");
+//                    break;
+//                    }
+                    System.out.println("Input new product's info: ");
+                    System.out.println("Name: "); String newName = scanner.nextLine();
+                    System.out.println("Description: "); String newDescription = scanner.nextLine();
+                    System.out.println("Category: "); String newCategory = scanner.nextLine();
+                    BigDecimal newCost = BigDecimal.valueOf(getDoubleInput("Cost (with \".\" as a decimal separator): ", scanner));
+                    long newAmount = (long) getDoubleInput("Amount (integer - the fractional part will be discarded): ", scanner);
+                    productDao.save(new Product(newName, newDescription, newCategory, newCost, newAmount));
+                    ifContinueMenu(scanner);
+                    break;
+                case "7":
+                    System.out.println("Input name of product You want to update: ");
+                    name = scanner.nextLine();
+                    new ArrayList<>(productDao.getByNameSubstring(name)).forEach(System.out::println);
```

Comment:
why new ArrayList()?

Reply:
yes, this is redundantly. There an array is created in productDao